src/main/ruby/core/posix.rb

```diff
@@ -54,15 +55,26 @@ def self.attach_function(name, argument_types, return_type)
   attach_function :setgid, [:gid_t], :int
   attach_function :setuid, [:uid_t], :int
 
+  attach_function :getrlimit, [:int, :pointer], :int
+  attach_function :setrlimit, [:int, :pointer], :int
+
+  attach_function :getenv_native, :getenv, [:string], :string
+  def self.getenv(name)
+    value = getenv_native(name)
+    if value.nil?
+      nil
+    else
+      ptr = Rubinius::FFI::Pointer.new(Truffle::Interop.as_pointer(value))
+      ptr.read_string_to_null
```

Comment:
We can finally read bytes as bytes without going through a Java String!
cc @nirvdrum